Mathlib/Order/Basic.lean

```diff
@@ -243,15 +223,42 @@ theorem ge_iff_eq (h : a ≤ b) : b ≤ a ↔ a = b :=
 end LE.le
 
 -- See Note [decidable namespace]
-protected theorem Decidable.le_iff_eq_or_lt [DecidableLE α] : a ≤ b ↔ a = b ∨ a < b :=
+namespace Decidable
+variable [DecidableLE α]
+
+protected theorem eq_or_lt_of_le (hab : a ≤ b) : a = b ∨ a < b :=
+  (lt_or_eq_of_le hab).symm
+
+protected theorem le_iff_lt_or_eq : a ≤ b ↔ a < b ∨ a = b :=
+  ⟨lt_or_eq_of_le, le_of_lt_or_eq⟩
+
+protected theorem le_iff_eq_or_lt : a ≤ b ↔ a = b ∨ a < b :=
   Decidable.le_iff_lt_or_eq.trans or_comm
 
+end Decidable
+
+theorem le_iff_lt_or_eq : a ≤ b ↔ a < b ∨ a = b :=
+  open scoped Classical in Decidable.le_iff_lt_or_eq
+
+theorem ge_iff_gt_or_eq : a ≤ b ↔ a < b ∨ b = a := propext eq_comm ▸ le_iff_lt_or_eq
 theorem le_iff_eq_or_lt : a ≤ b ↔ a = b ∨ a < b := le_iff_lt_or_eq.trans or_comm
+theorem ge_iff_eq_or_gt : a ≤ b ↔ b = a ∨ a < b := ge_iff_gt_or_eq.trans or_comm
+
+theorem lt_or_eq_of_le : a ≤ b → a < b ∨ a = b := le_iff_lt_or_eq.mp
+theorem gt_or_eq_of_ge : a ≤ b → a < b ∨ b = a := ge_iff_gt_or_eq.mp
```

Comment:
I'm struggling to see how this follows the naming guide; is the naming process now:
* Read the conclusion, `a < b ∨ b = a`
* From left to right, this starts as `lt_or_...`
* Oh, the equality is not in alphabetical order; so lets start over with `gt_or_eq`
* Now read the hypothesis `a ≤ b`, but this is not `of_le` because we swapped the order due to the equality we visited earlier.

Can you make a zulip thread specifically about these renames? I suspect that many people will have not been following the `order_dual` thread, since the title suggests that's some implementation detail that someone doing mathematics won't care about.

But if what's actually happening is that the naming convention is changing, this should be more widely broadcast.

Reply:
Zulip thread: [#mathlib4 > naming convention for ≤ and < @ 💬](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/naming.20convention.20for.20.E2.89.A4.20and.20.3C/near/514545998)